src/core_plugins/metrics/server/lib/get_vis_data.js

```diff
@@ -0,0 +1,15 @@
+import _ from 'lodash';
+import getPanelData from './vis_data/get_panel_data';
+
+function getVisData(req) {
+  const reqs = req.payload.panels.map(getPanelData(req));
```

Comment:
It's mapping panels to a set of promises (which is requesting the data for each panel) then waiting for them to resolve then joining together the results in the response object. I'm changing `reqs` to `promises` to make that more clear.